http/src/v0/refs.rs

```diff
@@ -1,94 +1,617 @@
+                    let name = match m.remove("Name") {
+                        // not sure of this, not covered by tests, though these are only
+                        // present for multi-block files so maybe it's better to panic
+                        Some(Ipld::String(s)) if s == "/" => {
+                            unimplemented!("Slashes as the name of link")
+                        }
+                        Some(Ipld::String(s)) => Some(s),
+                        Some(x) => panic!(
+                            "Expected dag-pb2ipld \"Links[{}]/Name\" to be a string, got: {:?}",
+                            i, x
+                        ),
+                        // not too sure of this, this could be the index as string as well?
+                        None => unimplemented!(
+                            "Default name for dag-pb2ipld links, should it be index?"
+                        ),
+                    };
+
+                    Some((name, link))
+                }
+                x => panic!(
+                    "Expected dag-pb2ipld \"Links[{}]\" to be a map, got: {:?}",
+                    i, x
+                ),
+            }
+        })
+        .collect()
 }
 
 /// Handling of https://docs-beta.ipfs.io/reference/http/api/#api-v0-refs-local
 pub fn local<T: IpfsTypes>(
     ipfs: &Ipfs<T>,
 ) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
-    path!("refs" / "local")
+    warp::path!("refs" / "local")
         .and(with_ipfs(ipfs))
         .and_then(inner_local)
 }
 
 async fn inner_local<T: IpfsTypes>(ipfs: Ipfs<T>) -> Result<impl Reply, Rejection> {
-    let refs: Vec<Result<String, Error>> = ipfs
+    let refs = ipfs
         .refs_local()
         .await
         .map_err(StringError::from)?
         .into_iter()
         .map(|cid| cid.to_string())
-        .map(|refs| RefsResponseItem {
-            refs,
-            err: "".to_string(),
+        .map(|refs| Edge {
+            ok: refs.into(),
+            err: "".into(),
         })
         .map(|response| {
             serde_json::to_string(&response)
+                .map(|mut s| {
+                    s.push('\n');
+                    s
+                })
                 .map_err(|e| {
                     eprintln!("error from serde_json: {}", e);
                     HandledErr
                 })
-                .unwrap()
-        })
-        .map(|ref_json| Ok(format!("{}{}", ref_json, "\n")))
-        .collect();
+        });
 
     let stream = stream::iter(refs);
     Ok(warp::reply::Response::new(Body::wrap_stream(stream)))
 }
 
-#[derive(Debug)]
-struct HandledErr;
+#[cfg(test)]
+mod tests {
+    use super::{local, refs_paths, Edge, IpfsPath};
+    use futures::stream::TryStreamExt;
+    use ipfs::{Block, Ipfs};
+    use libipld::block::{decode_ipld, validate};
+    use libipld::cid::Cid;
+    use std::collections::HashSet;
+    use std::convert::TryFrom;
+
+    #[tokio::test]
+    async fn test_inner_local() {
```

Comment:
this could be `local_refs` since it's no longer testing just the inner_local

Reply:
You're free to do what you like with the tests, that one is more of a stub anyway